auto_align/autofocus.py

```diff
@@ -92,3 +97,9 @@ def auto_align(measure_fn, should_stop_fn):
                 )
                 measurements = []
                 (step_size, measure_secs) = step_sizes.pop(0)
+
+if has_serial:
+    auto_align = auto_align1
+else:
+    def auto_align(*args, **kwargs):
+        return
```

Comment:
If I understand correctly, this is changing the error behavior for auto alignment if serial is not available.
Before, it would throw an error (and kill the program); now it will silently fail.

In general, I'm happy with not killing the program, but 2 comments:
1. this is related to auto-correlation (if not, can we address it in a separate PR?)
2. I don't think it's great to fail silently. Maybe we could instead add a method `auto_correlation_supported` (or similar) and disable the Button in the UI if it is not supported?

Reply:
My responses:
1. Not directly, it is added so the software can be tested without having to install uldaq etc., a separate PR is fine with me
2. I'm ok with making it more explicit, as explained, I just wanted to be able to test without installing everything, so making it more explicit is fine with me.

Reply:
I've updated the code so it now sets has_align based on if it can import autoalign, and that will enable/disable the auto align button accordingly.